plugins/unmanaged_files/unmanaged_files_inspector.rb

```diff
@@ -213,6 +213,18 @@ def inspect(filter, options = {})
 
     scope.scope_file_store = file_store_tmp
 
+    helper = MachineryHelper.new(@system)
+    if helper.can_help?
+      helper.inject_helper
+      helper.run_helper(scope)
+      scope.extracted = false
+      @description["unmanaged_files"] = scope
```

Comment:
A `remove_helper` call is missing here, no?

Since the helper is ultimately not restricted to one scope I wonder if a scheme where the `System` object holds a reference to its helper would not be more suitable.

How I would imagine it without putting too much thought into it is that the `InspectTask` takes care of installing the helper when possible and removing it after the inspection. The inspectors get the system instance passed anyway, so they could just ask the system if it has a helper or not and decide what to do accordingly:

```ruby
if system.helper
  system.helper.run(scope)
  ...
else
  run_inspection(filter, options, do_extract, file_store_tmp, file_store_final, scope)
end
```